Source/AliveLibAO/TimedMine.cpp

```diff
@@ -184,6 +187,155 @@ __int16 TimedMine::VTakeDamage_408B90(BaseGameObject* pFrom)
+    FP hitX = {};
+    PathLine *pLine = nullptr;
+    field_1B8 |= 1;
+    if (sCollisions_DArray_504C6C->RayCast_40C410(
+        field_A8_xpos,
+        field_AC_ypos - FP_FromInteger(20),
+        field_A8_xpos,
+        field_AC_ypos + FP_FromInteger(20),
+        &pLine,
+        &hitX,
+        &hitY,
+        (field_BC_sprite_scale != FP_FromDouble(0.5)) ? 7 : 112))
+    {
+        if (pLine->field_8_type == 32 || pLine->field_8_type == 36)
+        {
+            if (ObjListPlatforms_50766C)
+            {
+                for (int i = 0; i < ObjListPlatforms_50766C->Size(); i++)
+                {
+                    BaseGameObject* pObj = ObjListPlatforms_50766C->ItemAt(i);
+                    if (!pObj)
+                    {
+                        break;
+                    }
+
+                    if (pObj->field_4_typeId == Types::eLiftPoint_51)
+                    {
+                        PSX_RECT pObjRect = {};
+                        auto pLiftPoint = static_cast<LiftPoint*>(pObj);
+                        pLiftPoint->VGetBoundingRect_418120(&pObjRect, 1);
+                        if (FP_GetExponent(field_A8_xpos) > pObjRect.x &&
+                            FP_GetExponent(field_A8_xpos) < pObjRect.w &&
+                            FP_GetExponent(field_AC_ypos) < pObjRect.h)
+                        {
+                            field_F8_pLiftPoint = pLiftPoint;
+                            pLiftPoint->VAdd(this);
+                            ++field_F8_pLiftPoint->field_C_refCount;
+                            return;
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+void TimedMine::VUpdate()
+{
+    VUpdate_408760();
+}
+
+void TimedMine::VUpdate_408760()
+{
+    auto pPlatform = static_cast<LiftPoint*>(field_F8_pLiftPoint);
+    if (Event_Get_417250(kEventDeathReset_4))
+    {
+        field_6_flags.Set(BaseGameObject::eDead_Bit3);
+    }
+
+    if (!(field_1B8 & 1))
+    {
+        StickToLiftPoint_408CA0();
+    }
+
+    if (pPlatform && pPlatform->OnAnyFloor())
+    {
+        field_D4_collection_rect.x = field_A8_xpos - ScaleToGridSize_41FA30(field_BC_sprite_scale) / FP_FromInteger(2);
+        field_D4_collection_rect.y = field_AC_ypos - ScaleToGridSize_41FA30(field_BC_sprite_scale);
+        field_D4_collection_rect.w = field_A8_xpos + ScaleToGridSize_41FA30(field_BC_sprite_scale) / FP_FromInteger(2);
+        field_D4_collection_rect.h = field_AC_ypos;
+    }
+    if (field_10C_armed == 1)
+    {
+        if (static_cast<int>(gnFrameCount_507670) > (field_1B4 + field_1B0))
+        {
+            field_1B0 = gnFrameCount_507670;
+            auto direction = gMap_507BA8.GetDirection(
+                field_B2_lvl_number,
+                field_B0_path_number,
+                field_A8_xpos,
+                field_AC_ypos);
+            SFX_Play_43AED0(3u, 50, direction);
+
+            if (((field_114_timer - gnFrameCount_507670) & ~7) >= 18 * 8)
```

Comment:
add a comment saying its rounding to multiples of 8 so we don't have to figure this out again?

Reply:
Added